chain/network/src/codec.rs

```diff
@@ -74,12 +74,63 @@ pub fn bytes_to_peer_message(bytes: &[u8]) -> Result<PeerMessage, std::io::Error
     PeerMessage::try_from_slice(bytes)
 }
 
+fn peer_id_type_field_len(enum_var: u8) -> Option<usize> {
+    // 1 byte for enum variant, then some number depending on the
+    // public key type
+    match enum_var {
+        0 => Some(1 + 32),
+        1 => Some(1 + 64),
+        _ => None,
+    }
+}
+
+pub fn is_forward_tx(bytes: &[u8]) -> bool {
+    // PeerMessage::Routed variant == 13
+    if bytes[0] == 13 {
+        let target_field_len = if bytes[1] == 0 {
+            match peer_id_type_field_len(bytes[2]) {
+                Some(l) => l,
+                None => {
+                    return false;
+                }
+            }
+        } else if bytes[1] == 1 {
+            // PeerIdOrHash::Hash is always 32 bytes
+            32
+        } else {
+            return false;
+        };
+        let author_variant_idx = 2 + target_field_len;
+        let author_field_len = match peer_id_type_field_len(bytes[author_variant_idx]) {
+            Some(l) => l,
+            None => {
+                return false;
+            }
+        };
+        let signature_variant_idx = author_variant_idx + author_field_len;
+        let signature_field_len = match bytes[signature_variant_idx] {
+            0 => 1 + 64,
+            1 => 1 + 65,
+            _ => {
+                return false;
+            }
+        };
+        let ttl_idx = signature_variant_idx + signature_field_len;
+        let message_body_idx = ttl_idx + 1;
+
+        bytes[message_body_idx] == 1
```

Comment:
What field does this one correspond to?

Reply:
This byte is the enum variant of `RoutedMessageBody`.